### PR TITLE
Unify timeouts in tests

### LINKED ISSUE #1706: Unify timeout values used in tests
We have lots of tests that use timeout values, eg. to check that a given operation timeouts as expected or completes a given amount of work before a certain amount of time. Unfortunately, in CI those tests tend to fail some of the time, because say 1% of the time everything is super slow.

In response to this, we went from 0.001 to 0.01 and sometimes 0.1 in the places that failed the most often (#1554 and #1496 are recent examples). But there are still some tests that fail because they were not updated. Since there is no test that is inherently slow in the test suite that I'm aware of, I think it would be nice to share a few timeout values (ideally two) in all tests, so that we can:

 1. bump them all at once if needed
 2. use smaller values in local environments, and larger values in CI

### PR DESCRIPTION
**Pull request by pquentin**
Please review #1712 first, as I included it in this pull request to avoid conflicts in the future.

Closes #1706

### CODE REVIEW

File: test/__init__.py [hunk 1]
```diff
@@ -28,6 +28,16 @@
...
 INVALID_SOURCE_ADDRESSES = [("192.0.2.255", 0), ("2001:db8::1", 0)]
 
+# We use timeouts in three different ways in our tests
+#
+# 1. To make sure that the operation timeouts, we can use a short timeout.
+# 2. To make sure that the test does not hang even if the operation should succeed, we
+#    want to use a long timeout, even more so on CI where tests can be really slow
+# 3. To test our timeout logic by using two different values, eg. by using different
+#    values at the pool level and at the request level.
+SHORT_TIMEOUT = 0.001
```
Comment (sethmlarson) [inline test/__init__.py]: Looks like Travis (which is faster than AppVeyor usually) timed out on this `0.001` value. Maybe we need to have two separate values for the `SHORT_TIMEOUT` as well?

File: test/appengine/test_gae_manager.py [hunk 2]
```diff
@@ -46,7 +47,7 @@ def setup_method(self, method):
     def test_exceptions(self):
         # DeadlineExceededError -> TimeoutError
         with pytest.raises(urllib3.exceptions.TimeoutError):
-            self.pool.request("GET", "/sleep?seconds=0.005", timeout=0.001)
+            self.pool.request("GET", "/sleep?seconds=0.005", timeout=SHORT_TIMEOUT)
```
Comment (sethmlarson) [inline test/appengine/test_gae_manager.py]: If we implement the above comment, can we have the `seconds=X` call be dynamic? Maybe 2 * SHORT_TIMEOUT?

File: test/__init__.py (hunk 1 above)
Comment (pquentin) [reply test/__init__.py]: Oh, `test_ssl_read_timeout` failed, I had not noticed that, sorry. I think the better fix is to switch to `LONG_TIMEOUT` here. Here's my reasoning.

In this test, we:
 1. read the headers -- this should not timeout, so using `LONG_TIMEOUT` makes sense
 2. read the body -- this *should* timeout, so using `SHORT_TIMEOUT` makes sense

However, both timeouts are read timeouts and are controlled by a single value, so we have to compromise. Switching to `LONG_TIMEOUT` is only a compromise for this test, while bumping `SHORT_TIMEOUT` would be a compromise for all tests.

What do you think?

File: test/appengine/test_gae_manager.py (hunk 2 above)
Comment (pquentin) [reply test/appengine/test_gae_manager.py]: Ah yes, great catch! I think I'll continue to use 5 * SHORT_TIMEOUT

**Comment by pquentin** [comment]
Rebased on top of master and added two commits for your first two remarks.

**Comment by pquentin** [comment]
I broke the lint, sorry. Will fix as soon as I can.

**Comment by pquentin** [comment]
Closing/reopening to see if I just got lucky or not.

**Comment by pquentin** [comment]
The AppVeyor failure is due to my test from #1715, I'll fix it later. The Travis failure is due to 100ms not being enough, I bumped LONG_TIMEOUT to 500ms.

**Comment by pquentin** [comment]
Closing/reopening because it was a good idea the last time. :)

**Comment by pquentin** [comment]
One last try.

File: test/__init__.py (hunk 1 above)
Comment (sethmlarson) [reply test/__init__.py]: +1 LONG_TIMEOUT works for that test.

**Pull request merged**